frontend/cypress/integration/common/sidecar_injection.ts

```diff
@@ -239,32 +239,64 @@ When('I visit the overview page', function () {
 });
 
 When('I override the default automatic sidecar injection policy in the namespace to enabled', function () {
-  cy.get('[data-test=overview-type-LIST]').should('be.visible').click();
-  cy.get(`[data-test=VirtualItem_${this.targetNamespace}] button`).should('be.visible').click();
-  cy.get(`[data-test=enable-${this.targetNamespace}-namespace-sidecar-injection]`).should('be.visible').click();
-  cy.get('[data-test=confirm-traffic-policies]').should('be.visible').click();
-  ensureKialiFinishedLoading();
+  cy.request('GET', '/api/status').should(response => {
+    expect(response.status).to.equal(200);
+    const isMaistra = response.body.istioEnvironment.isMaistra;
+
+    cy.get('[data-test=overview-type-LIST]').should('be.visible').click();
+    cy.get('[data-test=overview-type-LIST]').should('be.visible').click();
+    cy.get(`[data-test=VirtualItem_${this.targetNamespace}] button`).should('be.visible').click();
+    // Maistra does not support namespace-level injection
+    if (isMaistra) {
+      cy.get(`[data-test=enable-${this.targetNamespace}-namespace-sidecar-injection]`).should('not.exist');
+    } else {
```

Comment:
These actually seem like two separate tests. You don't need to include this with this change but it'd be nice to start using tags to include/exclude certain tests based on the env. We'll need to do this for multi-cluster tests and probably other things as well.

Reply:
Yes, at the moment just trying to get the stuff passing so it's not blocking 2.4.  With 3.0 it's possible things will align with upstream and we won't have the disparity.